cabal-install/Distribution/Client/Setup.hs

```diff
@@ -2949,3 +2951,23 @@ relevantConfigValuesText :: [String] -> String
 relevantConfigValuesText vs =
      "Relevant global configuration keys:\n"
   ++ concat ["  " ++ v ++ "\n" |v <- vs]
+
+
+-- ------------------------------------------------------------
+-- * Commands to support show-build-info
+-- ------------------------------------------------------------
+
+showBuildInfoCommand :: CommandUI (BuildFlags, BuildExFlags)
+showBuildInfoCommand = parent {
+    commandDefaultFlags = (commandDefaultFlags parent, mempty),
+    commandOptions      =
+      \showOrParseArgs -> liftOptions fst setFst
+                          (commandOptions parent showOrParseArgs)
+                          ++
+                          liftOptions snd setSnd (buildExOptions showOrParseArgs)
+  }
+  where
+    setFst a (_,b) = (a,b)
```

Comment:
@phadej don't we have `_1`/`_2` lenses for this somewhere? :-)

Reply:
Such custom setters are used in all `Cmd*` files. So, I either doubt it, or I can do a subsequent pr to replace that for every Cmd.